crates/primitives/src/chain_spec.rs

```diff
@@ -120,62 +115,97 @@ impl ChainSpec {
         self.genesis_hash
     }
 
-    /// Returns the supported hardforks and their fork block numbers
-    pub fn hardforks(&self) -> &BTreeMap<Hardfork, BlockNumber> {
+    /// Returns the supported hardforks and their [ForkKind]
+    pub fn hardforks(&self) -> &BTreeMap<Hardfork, ForkKind> {
         &self.hardforks
     }
 
-    /// Get the first block number of the hardfork.
+    /// Get the first block number of the given hardfork. This method returns `None` for
+    /// timestamp-based forks and if the merge netsplit block is not known (when the given fork
+    /// is TTD-based)
     pub fn fork_block(&self, fork: Hardfork) -> Option<BlockNumber> {
+        self.hardforks.get(&fork).and_then(|kind| match kind {
+            ForkKind::Block(block_number) => Some(*block_number),
+            ForkKind::TTD(block_number) => *block_number,
+            _ => None,
+        })
+    }
+
+    /// Get the first block number of the hardfork.
+    pub fn fork_kind(&self, fork: Hardfork) -> Option<ForkKind> {
         self.hardforks.get(&fork).copied()
     }
 
     /// Returns `true` if the given fork is active on the given block
```

Comment:
should update docs here to reflect that this returns true if the fork is active on the given block / ttd / timestamp contained in the `ForkDiscriminant`